### PR TITLE
Hashnsalt

### PR DESCRIPTION
**Pull request by ak2253**

### CODE REVIEW

File: server/__init__.py [hunk 1]
```diff
@@ -22,3 +22,4 @@
 
 import server.models
 import server.routes
+import server.modules
```
Comment (dvkm) [inline server/__init__.py]: Don't think this is necessary?

File: server/modules/__init__.py [hunk 2]
```diff
@@ -0,0 +1 @@
+import server.modules.hash
```
Comment (dvkm) [inline server/modules/__init__.py]: Is this file necessary?

**Review by speshori — APPROVED** [review]

File: server/routes/pwd_login.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    query = db.session.query(User).filter(User.username == username).one()
+    return query.id
+
+@app.route("/api/login/password", methods=["POST"])
+def password_login():
+    try:
+        data = json.loads(request.data)
+        username = data['username']
+        password = data['password']
+        if(find_username is None or hash_login(get_pwd(username),password) is False):
```
Comment (dvkm) [inline server/routes/pwd_login.py]: add a space between a comma and the variable
`get_pwd(username), password)`

File: src/Login.tsx [hunk 4]
```diff
@@ -11,9 +12,81 @@ function googleLogin(googleUser: any) {
...
+    password: '',
+  });
+  function GoToSignUp() {
+    history.push('./signup');
+  }
+  function PerformLogin() {
+    const tusername = login.username.trim();
+    const tpassword = login.password.trim();
+
+    if (tusername === '' || tpassword === '') setLoginMessage('Please fill in both fields.');
```
Comment (dvkm) [inline src/Login.tsx]: Move setLoginMessage to the next line and put it inside the bracket

**Review by dvkm — CHANGES_REQUESTED** [review]

File: server/modules/__init__.py (hunk 2 above)
Comment (ak2253) [reply server/modules/__init__.py]: Its for incase more files are to be add into utils.

**Review by dvkm — APPROVED** [review]
